# """
# Задание 1.	Написать программу, которая будет складывать, вычитать,
# умножать или делить два числа. Числа и знак операции вводятся пользователем.
# После выполнения вычисления программа не должна завершаться, а должна
# запрашивать новые данные для вычислений. Завершение программы должно
# выполняться при вводе символа '0' в качестве знака операции. Если пользователь
# вводит неверный знак (не '0', '+', '-', '*', '/'), то программа должна
# сообщать ему об ошибке и снова запрашивать знак операции.
#
# Также сообщать пользователю о невозможности деления на ноль,
# если он ввел 0 в качестве делителя.
#
# Подсказка:
# Вариант исполнения:
# - условие рекурсивного вызова - введена операция +, -, *, / - ШАГ РЕКУРСИИ
# - условие завершения рекурсии - введена операция 0 - БАЗОВЫЙ СЛУЧАЙ
#
# Решите через рекурсию. В задании нельзя применять циклы.
#
# Пример:
# Введите операцию (+, -, *, / или 0 для выхода): +
# Введите первое число: 214
# Введите второе число: 234
# Ваш результат 448
# Введите операцию (+, -, *, / или 0 для выхода): -
# Введите первое число: вп
# Вы вместо трехзначного числа ввели строку (((. Исправьтесь
# Введите операцию (+, -, *, / или 0 для выхода):
# """


class SignError(Exception):
    pass


def calc_rec():
    sign = input('Введите операцию (+, -, *, / или 0 для выхода): ')
    try:
        if sign == '0':
            return
        elif sign not in ('/', '+', '-', '*'):
            raise SignError
        else:
            first_numb, second_numb = int(input('Введите первое число: ')), int(input('Введите второе число: '))
            if sign == '+':
                print(first_numb + second_numb)
            elif sign == '-':
                print(first_numb - second_numb)
            elif sign == '*':
                print(first_numb * second_numb)
            else:
                print(first_numb / second_numb)
            calc_rec()
    except SignError:
        print('Вы ввели недопустимую операцию, попробуйте ещё раз')
        calc_rec()
    except ZeroDivisionError:
        print('Деление на ноль невозможно, попробуйте ещё раз')
        calc_rec()
    except ValueError:
        print('Вы ввели число, отличное от int, попробуйте ещё раз')
        calc_rec()


if __name__ == '__main__':
    calc_rec()
